src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -92,6 +92,129 @@ def _get_symbol_table(node):
+        assert isinstance(reference, Reference), \
+            "Expecting Reference but found '{0}'." \
+            "".format(type(reference).__name__)
+        assert reference.symbol is array.symbol, \
+            "Expecting Reference symbol '{0}' to be the same as array " \
+            "symbol '{1}'.".format(reference.symbol.name, array.symbol.name)
+        assert isinstance(literal, Literal), \
+            "Expecting Literal but found '{0}'." \
+            "".format(type(literal).__name__)
+        assert literal.datatype == DataType.INTEGER, \
+            "Expecting integer datatype but found '{0}'." \
+            "".format(literal.datatype)
+        assert literal.value == str(dim), \
+            "Expecting literal value '{0}' to be the same as the current " \
+            "array dimension '{1}'.".format(literal.value, str(dim))
+    except AssertionError as excinfo:
+        raise InternalError("psyir/frontend/fparser2.py:check_bound():"
+                            + str(excinfo))
+
+
+def check_literal(node, dim, index, value):
+    '''Utility function to check that the supplied array has a literal at
+    dimension index "dim" and range index "index" with value
+    "value".
+
+    The step part of the range node has an integer literal with
+    value 1 by default.
+
+    Assumes that the node argument is an array and that the
+    supplied dimension index is a Range node and that the supplied
+    range index is valid.
+
+    :param array: the node to check.
+    :type array: :py:class:`pysclone.psyir.node.array`
+    :param int dim: the dimension index to check.
+    :param int index: the index of the range to check (0 is the \
+        lower bound, 1 is the upper bound).
+    :param int value: the expected value of the literal.
+
+    :raises InternalError: if the supplied argument does not have the \
+    required properties.
+
+    '''
+    try:
+        literal = node.children[dim-1].children[index]
+        assert isinstance(literal, Literal), \
+            "Expecting Literal but found '{0}'" \
+            "".format(type(literal).__name__)
+        assert literal.datatype == DataType.INTEGER, \
+            "Expecting integer datatype but found '{0}'." \
+            "".format(literal.datatype)
+        assert literal.value == str(value), \
+            "Expecting literal value '{0}' to be the same as '{1}'." \
+            "".format(literal.value, str(value))
+    except AssertionError as excinfo:
+        raise InternalError("psyir/frontend/fparser2.py:check_literal():"
+                            + str(excinfo))
+
+
+def check_range(my_range):
+    '''Utility function to check whether a Range object is equivalent to a
+    ":" in Fortran array notation. The PSyIR representation of "a(:)"
+    is "a(lbound(a,1):ubound(a,1):1). Therefore, for array a index 1,
+    the lower bound is compared with "lbound(a,1)", the upper bound is
+    compared with "ubound(a,1)" and the step is compared with 1.
```

Comment:
Please extend slightly to say that if everything is fine then this routine silently returns.

Reply:
Good idea.